docs/IE_DG/supported_plugins/GPU.md

```diff
@@ -122,6 +122,12 @@ When specifying key values as raw strings (that is, when using Python API), omit
 | `KEY_TUNING_MODE`     | `TUNING_DISABLED` <br /> `TUNING_CREATE` <br />  `TUNING_USE_EXISTING`            | `TUNING_DISABLED` | Disable inference kernel tuning     <br /> Create tuning file (expect much longer runtime)  <br />         Use an existing tuning file. **Deprecated**. Will be removed in the next release |
 | `KEY_TUNING_FILE`     | `"<filename>"`                  | `""`              | Tuning file to create / use. **Deprecated**. Will be removed in the next release |
 
+## Quering GPU specific metric keys
+* MAX_BATCH_SIZE : returns maximum batch size for a given network which is not only executable but also does not lose performance due to the memory swap impact. The example code to set the required and optional configs for this metic is available in the following snippet:
```

Comment:
Pls add that the returned value is not aligned to any power or multiple of (e.g. 2).

Reply:
also that by default the max_batch is estimated for the total device memory and it is responsibility of the user to track (and hint the function with) the overall footprint (potentially including other processes).

Reply:
I think you are mentioning about "MEMORY_STATISTIC" metric to that purpose, but it cannot provide "const size"  and "general tensor size". So there is no possible scenario that the user can directly use that. Also the memory reusing mechanism matters for the estimation, which is not trivial for user to take into consideration.

Reply:
I changed the behavior to use "avialable memory size" which is [max_device_mem] - [ocupied_usm_device_mem_size] utilizing the MEMORY_STATISTICS query.